modules/chapter2/pages/rbac.adoc

```diff
@@ -0,0 +1,138 @@
+
+RBAC on RHDH is built on top of the *Permissions* framework in Backstage, which defines RBAC policies in code. Rather than define policies in code, the RHDH RBAC feature allows you to define policies in a declarative fashion using a simple CSV based format.
+
+To apply RBAC in RHDH, you need to do two things:
+
+. Configure an authenticated user as a *Policy Administrator*
+. Configure policies in a CSV file and import it into RHDH
+
+## Lab: Configure Policy Administrators
+
+NOTE: The main reason to declare policy administrators is to allow a certain limited number of authenticated users to invoke the RBAC REST API. The actual policies are defined in a separate CSV file and referenced in the `app-config-rhdh` ConfigMap. This can be done by OpenShift platform or cluster administrators who have access to the namespace where RHDH is running.
+
+. To declare users as policy administrators, add the following `permission` attribute to the `app-config-rhdh` ConfigMap (At the same indentation level as the *app* and below the *catalog* attribute). Ensure that you added a valid authenticated user for the `admin.users` attribute:
++
+[subs=+quotes]
+----
+catalog:
+      providers:
+        githubOrg:
+          default:
+            id: development
+            orgUrl: ${GITHUB_ORG_URL}
+*permission:
+      enabled: true
+      rbac:
+        admin:
+          users:
+            - name: user:/default/rsriniva*
+----
+
+. Re-deploy the `rhdh` helm chart for the changes to take effect.
+
+. Sign out from the existing RHDH session and login again using the GitHub ID of the declared policy administrator.
+
+. Navigate to the `Catalog` page in RHDH, and click `CREATE`. Note that you are not allowed to create new components.
+
+. Next, try navigating to the `API` page and click on `REGISTER EXISTING API`. You will be shown an `ERROR 404` page. With RBAC enabled, most features are disabled by default. You need to explicitly enable permissions to resources in RHDH.
++
+image::404-error-rbac.png[title=Not allowed to register APIs]
+
+## Lab: Declarative Policy Definition in CSV Files
+
+To selectively allow RHDH features, you need to declare `roles` with allowed policies, and then assign these roles to users or groups.
+
+. Edit the `app-config-rhdh` ConfigMap, and add a reference to the policy CSV file under the `rbac` attribute that you enabled previously:
++
+[subs=+quotes]
+----
+catalog:
+      providers:
+        githubOrg:
+          default:
+            id: development
+            orgUrl: ${GITHUB_ORG_URL}
+permission:
+      enabled: true
+      rbac:
+        *policies-csv-file: ./rbac/rbac-policy.csv*
+        admin:
+          users:
+            - name: user:/default/rsriniva
```

Comment:
this should probably read user:default/rsriniva
so the first / after the colon needs to be removed?

Reply:
fixed